main.tf

```diff
@@ -10,10 +10,10 @@ locals {
   network_firewall_policies              = local.enabled && length(var.network_firewall_policies) > 0 ? { for policy in flatten(var.network_firewall_policies) : policy.name => policy } : {}
 
 
-  logging_config_firehose_arn     = { logDestinationConfigs : [var.firehose_arn], redactedFields : [{ redactedFieldType : "SingleHeader", redactedFieldValue : "Cookies" }, { redactedFieldType : "Method" }] }
-  logging_config_firehose_enabled = { logDestinationConfigs : [join("", aws_kinesis_firehose_delivery_stream.firehose_stream.*.id)], redactedFields : [{ redactedFieldType : "SingleHeader", redactedFieldValue : "Cookies" }, { redactedFieldType : "Method" }] }
+  logging_config_firehose_arn     = jsonencode({ logDestinationConfigs : [var.firehose_arn], redactedFields : [{ redactedFieldType : "SingleHeader", redactedFieldValue : "Cookies" }, { redactedFieldType : "Method" }] })
```

Comment:
do we need to `jsonencode` it here?
we already `jsonencode` the whole thing https://github.com/cloudposse/terraform-aws-firewall-manager/blob/master/waf_v2.tf#L41

Reply:
We need the jsonencode for the ternary condition to have the same object types. TF unfortunately doesn't allow different typed objects in the conditional statement, even if its object object (with different number of fields. which is the case here from 2 fields `logDestinationConfigs` and `redactedFields` vs `{}`)

Reply:
@Benbentwo I believe this change results in a mis-formatted `managedServiceData` block. There're too many backslashes, around `loggingConfiguration` elements, it looks like this:

\"loggingConfiguration\":\"{\\\"logDestinationConfigs\\\":[\\\"<stream_arn>\\\"],\\\"redactedFields\\\":[{\\\"redactedFieldType\\\":\\\"SingleHeader\\\",\\\"redactedFieldValue\\\":\\\"Cookies\\\"}

and gets rejected by AWS API. This is with TF 1.0.9 and AWS Provider v3.73.0. Removing `jsonencode` solved the issue.